docs/02-concepts/01-data_structures/01-merkle_patricia_trie.md

```diff
@@ -0,0 +1,29 @@
+# Merkle Patricia Trie
+
+The Merkle Patricia Trie currently used in Ethereum is a modified fusion of the merkle tree and [patricia trie](https://en.wikipedia.org/wiki/Radix_tree) (also known as a radix tree or prefix tree).
+
+It turns out that merkle trees are great for verifying a large list of static data elements, but not so great for storing the Ethereum world state which changes frequently. With a merkle tree, a single change requires rehashing the entire tree. For 1 million data elements in a simple binary merkle tree, this would require $\approx$ 1 million hash operations. In reality, the Ethereum world state consists of much more than 1 million data elements. 
```

Comment:
Can you elaborate on what you mean here? If I change a single leaf in a merkle tree, I'd imagine that some parts of the tree do not change and so, would not require rehashing?

Reply:
Ahh you are right, have updated this paragraph in a new commit